### PR TITLE
Reduce osu editor allocations with many objects selected

### PR DESCRIPTION
**Pull request by EVAST9919**
|master|pr|
|---|---|
|master|pr|

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Edit/Blueprints/HitCircles/Components/HitCircleOverlapMarker.cs [hunk 1]
```diff
@@ -92,6 +89,10 @@ public override void UpdateFrom(HitCircle hitObject)
 
                 ring.Scale = new Vector2(1 + 0.1f * ringScale);
                 content.Alpha = 0.9f * (1 - alpha);
+
+                // TODO: should only update colour on skin/combo/object change.
```
Comment (EVAST9919) [inline osu.Game.Rulesets.Osu/Edit/Blueprints/HitCircles/Components/HitCircleOverlapMarker.cs:93]: This is a quick band-aid fix. We are still performing skin lookup on each frame, but now only for visible markers (previously for every alive one). Reworking this will require a bit of refactoring, which should probably be in it's own pr.

**Review by peppy — APPROVED** [review]
